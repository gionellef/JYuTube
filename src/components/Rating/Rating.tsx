import * as React from 'react';

import { Icon, Progress } from "semantic-ui-react";

import './Rating.scss';

interface RatingProps {
  likeCount?: number;
  dislikeCount?: number;
};

const Rating : React.FC<RatingProps> = (props) => {
  let progress = null;

  if (props.likeCount && props.dislikeCount) {
    const percent  = 100 * (props.likeCount / (props.likeCount + props.dislikeCount));

    progress = <Progress className='progress' percent={percent} size='tiny' />
  }
  
  return (
    <div className='rating'>
      <div className='thumbs-up'>
        <Icon name='thumbs up outline' />
        <span>{ props.likeCount }</span>
      </div>
      <div className='thumbs-down'>
        <Icon name='thumbs down outline' />
        <span>{ props.dislikeCount }</span>
      </div>
      { progress }
    </div>
  );
}

export default Rating;