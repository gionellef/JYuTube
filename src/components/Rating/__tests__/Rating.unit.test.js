import React from 'react';

import { shallow } from 'enzyme';

import { Rating } from '../Rating';

describe('Rating', () => {
  test('renders Rating without props', () => {
    const wrapper = shallow(
      <Rating />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('renders Rating with props likes and dislikes', () => {
    const wrapper = shallow(
      <Rating likeCount={1000} dislikeCount={100} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});