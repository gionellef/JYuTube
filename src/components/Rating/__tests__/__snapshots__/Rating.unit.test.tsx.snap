// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders Rating with props likes and dislikes 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs up outline"
    />
    <span>
      1000
    </span>
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs down outline"
    />
    <span>
      100
    </span>
  </div>
  <Progress
    className="progress"
    percent={90.9090909090909}
    size="tiny"
  />
</div>
`;

exports[`Rating renders Rating without props 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs up outline"
    />
    <span />
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs down outline"
    />
    <span />
  </div>
</div>
`;
